### PR TITLE
feat(truncate): add user id for truncation

### PR DESCRIPTION
**Pull request by peterdeme**
Add user id for truncation

### CODE REVIEW

**Comment by ferhatelmas** [comment]
looks good, any test?

**Comment by peterdeme** [comment]
@ferhatelmas - added test

**Review by ferhatelmas — DISMISSED** [review]

File: tests/ChannelClientTests.cs [hunk 1]
```diff
@@ -151,10 +151,12 @@ public async Task TestChannelTruncateAsync()
             var originalChannel = await _channelClient.GetOrCreateAsync(_channel.Type, _channel.Id, new ChannelGetRequest { State = true });
             originalChannel.Messages.Should().NotBeEmpty();
 
-            await _channelClient.TruncateAsync(_channel.Type, _channel.Id);
+            await _channelClient.TruncateAsync(_channel.Type, _channel.Id, new TruncateOptions { UserId = _user2.Id });
 
             var afterTruncateChannel = await _channelClient.GetOrCreateAsync(_channel.Type, _channel.Id, new ChannelGetRequest { State = true });
             afterTruncateChannel.Messages.Should().BeEmpty();
+            // this isn't deployed to the test environment, so uncomment it a bit later:
```
[the remark this answers is not in the document]
Comment (ferhatelmas) [reply tests/ChannelClientTests.cs:159]: good to uncomment and merge

[the remark this answers is not in the document]
Comment (peterdeme) [reply tests/ChannelClientTests.cs:159]: well, i can't merge with failed checks in this repo :D otherwise I would :D

File: src/Models/Channel.cs [hunk 2]
```diff
@@ -50,6 +50,9 @@ public class Channel : CustomDataBase
 
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "last_message_at")]
         public DateTimeOffset? LastMessageAt { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "truncated_by")]
+        public User TruncatedBy { get; set; }
```
Comment (ferhatelmas) [inline src/Models/Channel.cs:55]: you can get `truncated_by_id` if it's a deleted user

Comment (peterdeme) [reply src/Models/Channel.cs:55]: The backend doesn't return TruncatedById:

https://github.com/GetStream/chat/blob/ba899d26f0ab98b0ad9b5be7f591a9aaa651c2eb/types/channel.go#L803=

Comment (ferhatelmas) [reply src/Models/Channel.cs:55]: seems we don't expose it due to failing enrichment, we should have done though in ideal world

**Review by ferhatelmas — APPROVED** [review]

**Pull request merged**